superx/information_extractor.py

```diff
@@ -22,133 +22,139 @@ class info_extractor:
+                                                         'multiple_pages': True,
+                                                         'zip_link_prefix': None,
+                                                         'item_attr_name': 'Item',
+                                                         'is_weighted_attr_name': 'bIsWeighted',
+                                                         'item_date_format': '%Y-%m-%d'},
+                                           'matrix': {'store_name': 'victory',
+                                                      'url': 'http://matrixcatalog.co.il/NBCompetitionRegulations.aspx',
+                                                      'multiple_pages': False,
+                                                      'zip_link_prefix': 'http://matrixcatalog.co.il/',
+                                                      'item_attr_name': 'Product',
+                                                      'is_weighted_attr_name': 'BisWeighted',
+                                                      'item_date_format': '%Y-%m-%d'}
+                                           }
+
+    def run_extractor(self):
         """
-        This method web scrapes the urls in url_list and creates a set of the gzip file links.
-        The method then updates super_name and sends the set to parsing
+        This method starts the extraction process
         """
-        for url in self.url_list:
-            page = requests.get(url)
-            web_scrapper = BeautifulSoup(page.content, 'html.parser')
-            links_list = web_scrapper.find_all('a')
-            zip_links = set()
+        for key in self.super_specific_information:
+            self.current_super = self.super_specific_information[key]
+            url_list = [self.current_super['url']]
+
+            if self.current_super['multiple_pages']:
+                url_list = self.get_all_super_links()
+
+            self.get_zip_file_links(url_list)
 
-            for link in links_list:
-                if link.has_attr('href'):
-                    https = str(link.attrs['href'])
-                    if 'PriceFull' in https:
-                        zip_links.add(https)
+    def get_zip_file_links(self, url_list):
+        """
+        This method web scrapes the urls in url_list and creates a set of the gzip file links.
+        The method then sends the set to parsing
+        If connection to the url failed, moves on to next url
+        :param url_list: list of urls to extract zip files from
 
-            self.update_super_name(url)
-            self.info_parser(zip_links)
+        """
+        for url in url_list:
+            try:
+                page = requests.get(url)
+                web_scrapper = BeautifulSoup(page.content, 'html.parser')
+            except ConnectionError:
+                continue
```

Comment:
If there's a connection error, it will just ignore it and you won't know that you had a problem. Have you considered logging actions/problems?